engine/classes/ElggIcon.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+/**
+ * Entity icon class
+ */
+class ElggIcon extends ElggFile {
```

Comment:
I decided to add this right away, so we can do instance checks. We might decide to upgrade icons to entities in the future.